### DISCUSSION TITLE
Treasury Support

### DISCUSSION THREAD

**Discussion by BearLeonLight** [QuickShop-Community/QuickShop-Hikari#448] [Ideas]
Treasury's new economic communications platform
https://github.com/lokka30/Treasury/wiki/Comparison-to-other-APIs

**Comment by Ghost-chu** [comment]
Putting the program on hold indefinitely for now, after examining it I don't think Treasury's API design is good enough, Treasury over-engineered the economy API using complex methods that don't make it easy to use.
Also, the underlying layers of QuickShop-Hikari and QuickShop-Reremake are not designed for player and system account separation and need to be rewritten.

**Comment by Ghost-chu** [comment]
Added via commit: https://github.com/Ghost-chu/QuickShop-Hikari/commit/3a0e45a067457fe363b1d60c87493aa89406804f (experimental, not recommended)

Treasury's API is somewhat poorly designed to track successful completion of operations and is not recommended unless necessary.

**Reply by BearLeonLight** [reply]
Sorry, I also gave up using the app, not friendly enough.

Later I switched to TNE.

**Reply by Ghost-chu** [reply]
TNE much better, I personally don't recommend Treasury

**Reply by BearLeonLight** [reply]
I've been exploring TNE recently, and the developer responded with friendly support.
